### DISCUSSION TITLE
Make the library slightly more generic

### DISCUSSION THREAD

**Discussion by vauradkar** [Barugon/egui_file#54] [Ideas]
Thank you for writing and maintaining this library!

Will you be willing make the library more generic - something that can be used with filesystem like things?
My use case: working on an egui app that runs natively and as wasm webapp. When running in web, it transparently makes http requests (instead of fs system calls) to interact with remote filesystem to while maintaining same user experience.

Something I had in mind was like this https://github.com/Barugon/egui_file/compare/main...vauradkar:egui_file:main

**Comment by Barugon** [comment]
That looks pretty good. Why not make a PR for it?
